IdentityCore/src/requests/broker/MSIDSSOExtensionInteractiveTokenRequest.h

```diff
@@ -22,12 +22,28 @@
 // THE SOFTWARE.
 
 #import "MSIDInteractiveTokenRequest.h"
+#import "MSIDProviderType.h"
 
 NS_ASSUME_NONNULL_BEGIN
 
 API_AVAILABLE(ios(13.0))
 @interface MSIDSSOExtensionInteractiveTokenRequest : MSIDInteractiveTokenRequest
 
+- (nullable instancetype)initWithRequestParameters:(nonnull MSIDInteractiveRequestParameters *)parameters
+          oauthFactory:(nonnull MSIDOauth2Factory *)oauthFactory
+tokenResponseValidator:(nonnull MSIDTokenResponseValidator *)tokenResponseValidator
+            tokenCache:(nonnull id<MSIDCacheAccessor>)tokenCache
+                              accountMetadataCache:(nullable MSIDAccountMetadataCacheAccessor *)accountMetadataCache NS_UNAVAILABLE;
+
+- (nullable instancetype)initWithRequestParameters:(nonnull MSIDInteractiveRequestParameters *)parameters
+                                      oauthFactory:(nonnull MSIDOauth2Factory *)oauthFactory
+                            tokenResponseValidator:(nonnull MSIDTokenResponseValidator *)tokenResponseValidator
+                                        tokenCache:(nonnull id<MSIDCacheAccessor>)tokenCache
+                                      providerType:(MSIDProviderType)providerType
```

Comment:
how would the caller know the provider type?

Reply:
Moved provider type to OAuth Factories.